feat: copy material number and other identifiers in material list view

jakoballgaier opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description
<!-- 
Please describe your PR: 
- What does this PR introduce? 
- Does it fix a bug? 
- Does it add a new feature?
- Is it enhancing documentation?
-->
- added component to copy text to clipboard
- adjusted modal width
- split up order reference into own columns

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).

jakoballgaier committed:
feat: copy material number and other identifiers in material list view

- added component to copy text to clipboard
- adjusted modal width
- split up order reference into own columns

ReneSchroederLJ commented on frontend/src/components/ui/TextToClipboard.tsx:
```diff
@@ -0,0 +1,37 @@
+import { Button, Typography } from '@mui/material';
+
+type TextToClipboardProps = {
+    text?: string;
```
I don't think it makes sense that text is optional. The parent component should handle cases with empty text by showing a loader or simply not displaying the component depending on the circumstances.

ReneSchroederLJ commented on frontend/src/components/ui/TextToClipboard.tsx:
```diff
@@ -0,0 +1,37 @@
+
+type TextToClipboardProps = {
+    text?: string;
+    color?: string;
```
Instead of passing the color, I would add variants as a string union. For example "default" and "light". This prevents having to pass the same color value multiple times and instead enables autocomplete for the different variants.

ReneSchroederLJ commented on frontend/src/components/ui/TextToClipboard.tsx:
Whenever you create a new file it should start with a license header. You can see it used in other code files. In this case the header should be:

```javascript
/*
Copyright (c) 2025 Volkswagen AG
Copyright (c) 2025 Contributors to the Eclipse Foundation

See the NOTICE file(s) distributed with this work for additional
information regarding copyright ownership.

This program and the accompanying materials are made available under the
terms of the Apache License, Version 2.0 which is available at
https://www.apache.org/licenses/LICENSE-2.0.

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
License for the specific language governing permissions and limitations
under the License.

SPDX-License-Identifier: Apache-2.0
*/
```
Please always ensure that any new file is attributed by using a license header like this.

ReneSchroederLJ reviewed (changes requested):
Thank you for your contribution. Mostly works as intended I just have some minor improvements in mind.

Furthermore, since this will be your first PR in this project there are some tasks to be taken care of after you've made the requested adjustments.

Firstly, you should rebase your branch to main. Afterwards add a new entry to `CHANGELOG.md`. This will be the start of a new release version so you should copy the section of the previous version, including Known Knowns, remove the entries in Added, Changed and Fixed. The new Version number should be 4.1.1. Afterwards add your feature under `Added` with a placeholder for the PR link.

jakoballgaier committed:
fix: PR changes for copy material number and other identifiers in material list view

- added license header
- add variant to TextToClipboard component
- made text of TextToClipboard not nullable

jakoballgaier commented:
Thanks for your review, I adjusted the PR accordingly.

jakoballgaier committed:
Revert "feat: add environments for running frontends without containers"

This reverts commit 8b56a381fcf1fcaee871514a6d0315e6c8b8d1cc.

jakoballgaier committed:
Update CHANGELOG.md

jakoballgaier committed:
fix: PR changes for copy material number and other identifiers in material list view

- Fix copyright year
- Return null in TextToClipboard component if text is empty
- Add TextToClipboard component to material details header

jakoballgaier committed:
fix: PR changes for copy material number and other identifiers in material list view

- change material details header format

jakoballgaier committed:
Revert "feat: add environments for running frontends without containers"

This reverts commit 8b56a381fcf1fcaee871514a6d0315e6c8b8d1cc.

jakoballgaier committed:
fix: corrected expected heading in E2E test suite

jakoballgaier committed:
Merge branch 'main' into feat/copy-identifiers-in-material-list-view

jakoballgaier committed:
chore(frontend): bump version

jakoballgaier committed:
chore: fix version

jakoballgaier committed:
feat: add version to about license (#1067)

* feat: add version to about license

- added version placeholder
- get version from git tag or from package.json

* chore(frontend): bump version

* chore: fix version

jakoballgaier committed:
chore: fix package.lock version

jakoballgaier committed:
Merge branch 'main' into feat/copy-identifiers-in-material-list-view
